github/Requester.py

```diff
@@ -413,8 +421,8 @@ def __makeAbsoluteUrl(self, url):
             url = self.__prefix + url
         else:
             o = urlparse.urlparse(url)
-            assert o.hostname in [self.__hostname, "uploads.github.com"], o.hostname
-            assert o.path.startswith((self.__prefix, "/api/uploads"))
+            assert o.hostname in [self.__hostname, "uploads.github.com", "status.github.com"], o.hostname
```

Comment:
assert that the hostname can either be the existing API hostname, or to uploads or status API.
I'd prefer to not have this as hardcoded as it is here, but I don't want to overhaul even more; and the assertion at least helps in spotting coding errors.